### PR TITLE
[WIP] [NEVER TESTED] New models for Layer representation

### PR DESCRIPTION
**Pull request by Vegasq**
- Keep working on autogestion/corruption_tracker#26

### CODE REVIEW

File: claim/models.py [hunk 1]
```diff
@@ -107,3 +129,24 @@ def update_map(cls, json_data):
         for polygon in json_data["features"]:
             polygon['claim_count'] = polygons_dict.get(
                 str(polygon["properties"]["ID"]), 0)
+
+
+class Polygon(models.Model):
+    # TODO(vegasq) looks like here we only link Organization with Layer.
+    # probably we should move coordinates and layer to Organization
+    # model? And remove this connectivity layer.
+    organization = models.ForeignKey(Organization, null=True)
```
Comment (autogestion) [inline claim/models.py]: Polygon could cover not only building, but city district or county. So there would be polygons that are not organizations. Organization could be a subclass of Polygon

File: claim/models.py [hunk 2]
```diff
@@ -83,6 +83,28 @@ def __str__(self):
...
 
 
+class Layer(models.Model):
+    """
+        Some general representation of map.
+        f.e.
+            All schools of Lviv
+            All universities of Ukraine
+            etc.
+    """
```
Comment (autogestion) [inline claim/models.py:87]: Idea was to split organizations geographically. One GeoJSON for Kharkiv, another for Lviv. If we are going from file to db, Layer is equal to file. Insted of replacing the file, we change id of Layer in db query. Spliting by types could be done using org_type field. To get all schools of Lviv we could query db to get polygons with org_type = schools, and layer=Lviv, and layer_type=ORGANIZATION

To get agregation of claims by city districts (no organization polygons of map, just polygons which covers hole districts), we would query polygons with org_type =any, layer=Lviv,  layer_type=DISTRICT

In another words, Layer is the set of polygons, which would be returned as GeoJSON object to leaflet.js

File: claim/models.py [hunk 3]
```diff
@@ -83,6 +83,28 @@ def __str__(self):
...
+    LAYER_TYPES = (
+        (ORGANIZATION, _("Organization")),
+        (DISTRICT, _("District")),
+        (COUNTRY, _("Country")),
+    )
+
+    name = models.CharField(max_length=250)
+    layer_type = models.IntegerField(choices=LAYER_TYPES,
+                                     default=LAYER_TYPES[0][0])
+
```
Comment (autogestion) [inline claim/models.py:100]: layer_type = models.IntegerField(choices=LAYER_TYPES, default=ORGANIZATION)

File: claim/models.py (hunk 1 above)
Comment (autogestion) [inline claim/models.py]: Also there could be a case, when several organizations placed in one building. Better to keep them separate

Comment (Vegasq) [inline claim/models.py]: > Also there could be a case, when several organizations placed in one building. Better to keep them separate

Wow... i need to think on it. Looks like we need 

> organizations = models.ManyToMany(Organization)

Is there any way to represent multiple items incide of one polygon in leaflet?

Comment (autogestion) [inline claim/models.py]: leaflet have MultiPolygon
http://leafletjs.com/reference.html#multipolygon

Also, we could split building's polygon into smaller polygons of organizations in this building and place them close to each other, to cover all the building

**Comment by Vegasq** [comment]
Hey guys, i need yours help here. I stuck with:

ValueError: Cannot add "<Organization: Вечерняя школа №37>": the value for field "organization" is None

1. Just grab this patchset.
2. Remove exists DB.
3. ./manage.py migrate
4. ./manage.py initiate_db
...

Something strange with ManyToMany field.

**Comment by Vegasq** [comment]
Can you play a bit with latest patchset please? There is too much modifications.

**Pull request merged**